src/robot/variables/variableconverter.py

```diff
@@ -0,0 +1,22 @@
+#  Copyright 2008-2015 Nokia Networks
+#  Copyright 2016-     Robot Framework Foundation
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+from .typeconverters import TypeConverter
+
+def convert_variable_value(name, value, type_):
+        if not type_:
+            return value
+        converter = TypeConverter.converter_for(type_, allow_str_converter=True)
+        return converter.convert(name, value) if converter else value
```

Comment:
Why is a separate converted needed for variable values? Why cannot TypeConverter used directly?

Also, why has the `typeconverters` module been moved under `robot.variables`? I understand having it under `robot.running.arguments` may not be convenient when thinking about variable conversions, but in general I would say having it under variables is even worse. We need to discuss separately what's the best place for the code. Possibly generic parts could be moved under `robot.utils`. That could be done as a separate change before adding variable conversion.

Reply:
I split the PR after you request, this part of code is not anymore